HyperVExtension/src/DevSetupEngine/ConfigurationFileHelper.cs

```diff
@@ -43,30 +44,42 @@ public ApplicationResult(WinGet.ApplyConfigurationSetResult result)
 
     private WinGet.ConfigurationProcessor? _processor;
     private WinGet.ConfigurationSet? _configSet;
+    private PackageVersion _appInstallerVersion = new(0, 0, 0, 0);
 
     public ConfigurationFileHelper()
     {
     }
 
-    private static async Task InstallOrUpdateAppInstallerIfNeeded(IProgress<DevSetupEngineTypes.IConfigurationSetChangeData> progress)
+    private static PackageVersion GetAppInstallerVersion()
     {
-        const string AppInstallerPackageName = "Microsoft.DesktopAppInstaller";
         const string AppInstallerPackageFamilyName = "Microsoft.DesktopAppInstaller_8wekyb3d8bbwe";
-        const string AppInstallerStoreId = "9NBLGGH4NNS1";
 
-        var doInstall = false;
         Windows.Management.Deployment.PackageManager packageManager = new();
         var currentInstallerPackage = packageManager.FindPackagesForUser(string.Empty, AppInstallerPackageFamilyName).FirstOrDefault();
         if (currentInstallerPackage == null)
+        {
+            return new PackageVersion(0, 0, 0, 0);
+        }
+
+        return currentInstallerPackage.Id.Version;
+    }
+
+    private async Task InstallOrUpdateAppInstallerIfNeeded(IProgress<DevSetupEngineTypes.IConfigurationSetChangeData> progress)
+    {
+        const string AppInstallerPackageName = "Microsoft.DesktopAppInstaller";
+        const string AppInstallerPackageFamilyName = "Microsoft.DesktopAppInstaller_8wekyb3d8bbwe";
+        const string AppInstallerStoreId = "9NBLGGH4NNS1";

```

Comment:
Will this ID ever change?  If it might, maybe put this in the manifest?

Reply:
Who knows? It shouldn't, but we may discontinue this package completely in the future. We'd need to replace it with whatever replacement would be.
How manifest would help? We'd need to release an updated extension if we'll need to use a different package.

In general, it would be better to be able to query the object if functionality we need is supported. But here we also need to install or update the package, so we need those IDs anyway. 
Also, I don't see a clean way to query AppInstaller for functionality and I cannot test all previous versions to see whre they would fail.

For example, WinGet team recommended to try to use new processor id and if it fails use the old one. But that didn't work with at least one already publicly released version of AppInstaller. Instead of failing with an error it was hanging. 

So in this case I'd rather rely on version numbers I know than on unpredictable behavior of previous versions that I cannot test.

Reply:
> We'd need to release an updated extension if we'll need to use a different package.
I didn't consider this.  Even if this number was in a manifest of some kind a release still needs to happen if the Id changes.